mailbody.js

```diff
@@ -44,21 +50,27 @@ class Body extends events.EventEmitter {
     }
 
     parse_more (line) {
+        // Ensure we're working in buffers, for the tests (transaction should
+        // always pass buffers).
+        if (!Buffer.isBuffer(line)) line = Buffer.from(line);
+
         return this[`parse_${this.state}`](line);
     }
 
     parse_child (line) {
+        const line_string = line.toString();
+
         // check for MIME boundary
-        if (line.substr(0, (this.boundary.length + 2)) === (`--${this.boundary}`)) {
+        if (line_string.substr(0, (this.boundary.length + 2)) === (`--${this.boundary}`)) {
```

Comment:
Would this not be faster leaving as a Buffer and using `buf.compare()`?

Reply:
I was defaulting to doing all these comparisons as strings, since we end up needing string versions anyway, when we emit the boundary.  (I think it makes sense to emit strings for the boundaries, it just fits my expectations better than using buffers there.)  This more closely matches the existing code, so it feels less risky too.

It seems reasonable to me that we might gain some speed out of comparing as buffers and only converting to strings if necessary.  I don't have a good intuition of the performance characteristics around this.  If you think it'd be significant, I could try coding it up -- let me know how you feel about it.

Reply:
I'm sort of "meh" on it. Once you enable mail parsing in Haraka most performance gains are gone anyway.

Merging.